internal/events/aggregator.go

```diff
@@ -469,6 +469,12 @@ func (ag *aggregator) processMessage(ctx context.Context, manifest *core.BatchMa
 	case !dataAvailable:
 		l.Errorf("Message '%s' in batch '%s' is missing data", msgEntry.ID, manifest.ID)
 	default:
+		// Check the pin signer is valid for the message
+		action, err = ag.checkOnchainConsistency(ctx, msg, pin)
```

Comment:
This was the main goal of the PR - to move this check earlier instead of having it as part of `attemptMessageDispatch`.

However, since this call returns a `core.MessageAction`, it also made sense to refactor common handling of retry/wait actions up into this method as well, so consequently I split up `attemptMessageDispatch` a little bit to make that happen.

Reply:
Side note: `core.MessageAction` is definitely imperfect and can be slightly confusing. `ActionRetry` means "I also should have returned a non-nil err and you should treat that as retryable" (an err is ignored for any other action). `ActionConfirm` kind of means "looks good to me, go ahead to confirm or continue with the next step if there is one". There could be some further thought on how to handle success vs. the 3 distinct problem scenarios (wait/retry/reject).